webapp/src/plugin.jsx

```diff
@@ -6,35 +6,45 @@ import CreateIssueModal from 'components/modals/create_issue';
 
 import AttachCommentToIssuePostMenuAction from 'components/post_menu_actions/attach_comment_to_issue';
 import AttachCommentToIssueModal from 'components/modals/attach_comment_to_issue';
+import SetupUI from 'components/setup_ui';
 
 import PluginId from 'plugin_id';
 
 import reducers from './reducers';
 import {handleConnectChange, getConnected, handleInstanceStatusChange, getSettings} from './actions';
 
+export let setupUI;
+
+const setupUILater = (registry, store) => async () => {
```

Comment:
Instead of deferring everything. Could we just do `await getConnected()(store.dispatch, store.getState);` in the component? Then all the other components should update with the new info.

Reply:
@crspeller You mean call that in the dummy component? Off the top of my head, I'd rather keep all setup logic here, and let that dummy component just trigger the setup. But also, I'm not sure that would work because we don't want to show the UI components at all if the UI is turned off, and right now that only happens by not registering them in the first place.

Reply:
@crspeller You mean call that in the dummy component? Off the top of my head, I'd rather keep all setup logic here, and let that dummy component just trigger the setup. But also, I'm not sure that would work because we don't want to show the UI components at all if the UI is turned off, and right now that only happens by not registering them in the first place.

Reply:
Alright. I don't think this is worth worrying about at the 11th hour. It touches more broad architectural issues. If this works let's just merge it.

Reply:
@crspeller good point. This is just a fix for now. :)